lib/routes/account.js

```diff
@@ -995,6 +1063,11 @@ module.exports = (log, db, mailer, Password, config, customs, checkPassword, pus
             authAt: sessionToken.lastAuthAt()
           }
 
+          // Ensure that the correct verification method is set if one has been specified.
+          if (verificationMethod) {
+            response.verified = false
+            response.verificationMethod = verificationMethod
+          }
 
           if (! requestHelper.wantsKeys(request)) {
```

Comment:
Not related to your patch exactly, but I wonder if this should should now be `(! mustVerify)` or similar, now that having keys is not the only reason we might force a verification.